src/dmd/root/dcompat.h

```diff
@@ -24,6 +24,19 @@ struct DArray
         : length(length_in), ptr(ptr_in) { }
 };
 
+template<>
+struct DArray<char>
```

Comment:
Why isn't this using `DString` ?

Reply:
Probably because these changes predate `DString`

Reply:
On a second look, DString/DArray is missing some of the constructors / operator overloads. Should i add them?

Reply:
Sure. Which ones are missing ?

Reply:
```
DArray<char>& operator=(const char str[]) { ptr = str; length = strlen(str); }
```

Ignore the commant regarding the constructor, I forgot that C++ does not differentiate between `const char str[]` and `const char* str`

Reply:
Wait, even that operator overload is not required because of C++ implicit conversion ... feature. Sorry for the noise